guidance/src/guidance/guidance_worker.cpp

```diff
@@ -0,0 +1,183 @@
+                res.newState = plugins[i].activated;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void GuidanceWorker::system_alert_cb(const cav_msgs::SystemAlertConstPtr& msg)
+    {
+        gsm.onSystemAlert(msg);
+    }
+
+    void GuidanceWorker::plugin_discovery_cb(cav_msgs::Plugin msg)
+    {
+        // if plugin list does not have anything, add it to the list and return
+        if(plugins.size() == 0)
+        {
+            // check if a plugin required when it is newly discovered
+            msg.required = is_required_plugin(msg.name, msg.versionId);
+            plugins.push_back(msg);
+            return;
+        }
+        // if we have some entries in the list, check if the plugin is already in the record
+        size_t index = 0;
+        while(index < plugins.size())
+        {
+            // check version id to be safe, but in the normal case we do not allow same plugin with different version id existing together
+            if(plugins[index].name.compare(msg.name) == 0 && plugins[index].versionId.compare(msg.versionId) == 0)
+            {
+                // only availability can be changed by plugin itself
+                plugins[index].available = msg.available;
+                // this break will make sure that "index" variable stays at the location of matching entry if it exists
+                break;
+            }
+            ++index;
+        }
+        // index points to the end of pllugin list without a matching entry
+        if(index == plugins.size())
+        {
+            msg.required = is_required_plugin(msg.name, msg.versionId);
+            plugins.push_back(msg);
+        }
+    }
+
+    void GuidanceWorker::robot_status_cb(const cav_msgs::RobotEnabledConstPtr& msg)
+    {
+        gsm.onRoboticStatus(msg);
+    }
+
+    bool GuidanceWorker::guidance_acivation_cb(cav_srvs::SetGuidanceActiveRequest& req, cav_srvs::SetGuidanceActiveResponse& res)
+    {
+        // Translate message type from GuidanceActiveRequest to SetEnableRobotic
+        ROS_INFO_STREAM("Request for guidance activation recv'd with status " << req.guidance_active);
+        gsm.onSetGuidanceActive(req.guidance_active);
+        cav_srvs::SetEnableRobotic srv;
+        if (gsm.getCurrentState() == GuidanceStateMachine::ENGAGED) {
+            srv.request.set = cav_srvs::SetEnableRobotic::Request::ENABLE;
+            res.guidance_status = true;
+        } else {
+            srv.request.set = cav_srvs::SetEnableRobotic::Request::DISABLE;
+            res.guidance_status = false;
+        }
+        enable_client_.call(srv);
+        return true;
+    }
+
+    bool GuidanceWorker::spin_cb()
+    {
+        cav_msgs::GuidanceState state;
+        state.state = gsm.getCurrentState();
+        state_publisher_.publish(state);
+        return true;
+    }
+
+    int GuidanceWorker::run()
+    {
+        ROS_INFO("Initalizing guidance node...");
+        using std::placeholders::_1;
```

Comment:
I could be wrong but I don't think you actually need to define this. You can just say _1 in bind and this should resolve

Reply:
Yes. I do not have to use "using" keyword here. But in the function, I still need to specify the full namespace as ```std::placeholders::_1```.
But I updated it anyway because using "using" here is bad practice anyway.